cmd/kubeadm/app/phases/uploadcerts/uploadcerts.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package uploadcerts
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	clientset "k8s.io/client-go/kubernetes"
+	bootstraputil "k8s.io/cluster-bootstrap/token/util"
+	"k8s.io/klog"
+	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
+	kubeadmconstants "k8s.io/kubernetes/cmd/kubeadm/app/constants"
+	"k8s.io/kubernetes/cmd/kubeadm/app/util/apiclient"
+)
+
+//UploadCerts save certs needs to join a new control-plane on kubeadm-certs sercret.
+func UploadCerts(client clientset.Interface, certsDir string, certsToUpload []string, ownerToken kubeadmapi.BootstrapToken) error {
+	fmt.Printf("[uploadcerts] storing the certificates in ConfigMap %q in the %q Namespace\n", kubeadmconstants.KubeadmCertsSecret, metav1.NamespaceSystem)
+
+	secretData := map[string][]byte{}
+	for _, name := range certsToUpload {
+		cert, err := loadAndEncryptCert(certsDir, name)
+		if err != nil {
+			if os.IsNotExist(err) {
+				klog.V(2).Infof("[uploadcerts]cert don't exist: %s", name)
+				continue
+			}
+			return err
+		}
+		// "/" is not valid in a secret key
+		secretData[strings.Replace(name, "/", "-", -1)] = cert
+	}
+
+	ref, err := getOwnerReference(client, ownerToken)
+	if err != nil {
+		return err
+	}
+
+	return apiclient.CreateOrUpdateSecret(client, &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            kubeadmconstants.KubeadmCertsSecret,
+			Namespace:       metav1.NamespaceSystem,
+			OwnerReferences: ref,
+		},
+		Data: secretData,
+	})
+}
+
+func getOwnerReference(client clientset.Interface, ownerToken kubeadmapi.BootstrapToken) ([]metav1.OwnerReference, error) {
+	secretName := bootstraputil.BootstrapTokenSecretName(ownerToken.Token.ID)
+	secret, err := client.CoreV1().Secrets(metav1.NamespaceSystem).Get(secretName, metav1.GetOptions{})
+	if err != nil {
+		return nil, errors.Wrap(err, "error to get token reference")
+	}
+
+	gvk := schema.GroupVersionKind{Version: "v1", Kind: "Secret"}
+	ref := metav1.NewControllerRef(secret, gvk)
+	return []metav1.OwnerReference{*ref}, nil
+}
+
+func loadAndEncryptCert(certsDir, certName string) ([]byte, error) {
+	cert, err := ioutil.ReadFile(path.Join(certsDir, certName))
+	if err != nil {
+		return nil, err
+	}
+	//TODO: encrypt cert
```

Comment:
@neolit123 already did some research on this topic.
Here it is a link to an example using https://github.com/neolit123/kubeadm-getter/blob/master/pkg/crypto.go the target encryption method/key length

Reply:
something to note is that on these lines:
https://github.com/neolit123/kubeadm-getter/blob/master/pkg/crypto.go#L49
https://github.com/neolit123/kubeadm-getter/blob/master/pkg/crypto.go#L67

we need to use `key` directly where key is presumably generated by something like `createRandBytes(32)`

Reply:
Thanks, this helped a lot.